### ISSUE TITLE
1.2.12 when multiple users run realtime on the save device duplicate entry errors show up

### ISSUE THREAD

**Issue by bmfmancini** [Cacti/cacti#3791]
Hey Guys

Noticed this today 

if 2 users run realtime on the same device the following errors come up 

```
2020/09/04 15:15:49 - CMDPHP ERROR: A DB Exec Failed!, Error: Duplicate entry '122346-traffic_in-2020-09-04 15:15:49' for key 'PRIMARY'
2020/09/04 15:15:47 - CMDPHP ERROR: A DB Exec Failed!, Error: Duplicate entry '122346-traffic_in-2020-09-04 15:15:47' for key 'PRIMARY'
2020/09/04 15:15:40 - CMDPHP ERROR: A DB Exec Failed!, Error: Duplicate entry '122346-traffic_in-2020-09-04 15:15:40' for key 'PRIMARY'
2020/09/04 15:15:34 - CMDPHP ERROR: A DB Exec Failed!, Error: Duplicate entry '122346-traffic_in-2020-09-04 15:15:34' for key 'PRIMARY'
2020/09/04 15:15:29 - CMDPHP ERROR: A DB Exec Failed!, Error: Duplicate entry '122346-traffic_in-2020-09-04 15:15:29' for key 'PRIMARY'
2020/09/04 15:15:04 - CMDPHP ERROR: A DB Exec Failed!, Error: Duplicate entry '122346-traffic_in-2020-09-04 15:15:03' for key 'PRIMARY'
```

SQL Backtraces

```
2020/09/04 15:15:49 - CMDPHP SQL Backtrace: (/cmd_realtime.php[167]:db_execute(), /lib/database.php[213]:db_execute_prepared())
2020/09/04 15:15:47 - CMDPHP SQL Backtrace: (/cmd_realtime.php[167]:db_execute(), /lib/database.php[213]:db_execute_prepared())
2020/09/04 15:15:40 - CMDPHP SQL Backtrace: (/cmd_realtime.php[167]:db_execute(), /lib/database.php[213]:db_execute_prepared())
2020/09/04 15:15:34 - CMDPHP SQL Backtrace: (/cmd_realtime.php[167]:db_execute(), /lib/database.php[213]:db_execute_prepared())
2020/09/04 15:15:29 - CMDPHP SQL Backtrace: (/cmd_realtime.php[167]:db_execute(), /lib/database.php[213]:db_execute_prepared())
2020/09/04 15:15:04 - CMDPHP SQL Backtrace: (/cmd_realtime.php[167]:db_execute(), /lib/database.php[213]:db_execute_prepared())
```

the errors stop when 1 of the users stop realtime

**Comment by TheWitness** [comment]
Are you users using the guest account, or logged in as the same user?

**Comment by TheWitness** [comment]
Make sure your table create statement looks like this:

```sql
CREATE TABLE `poller_output_realtime` (
  `local_data_id` mediumint(8) unsigned NOT NULL DEFAULT 0,
  `rrd_name` varchar(19) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',
  `time` timestamp NOT NULL DEFAULT '0000-00-00 00:00:00',
  `output` text COLLATE utf8mb4_unicode_ci NOT NULL,
  `poller_id` varchar(256) COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '1',
  PRIMARY KEY (`local_data_id`,`rrd_name`,`time`,`poller_id`),
  KEY `poller_id` (`poller_id`(191)),
  KEY `time` (`time`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci ROW_FORMAT=DYNAMIC;
```

**Comment by TheWitness** [comment]
Yea, you just have a bad primary key.  This was resolved in 1.2.5, so you must have done some upgrade, hacked around some issues you had with it and moved on.  Just ensure that your poller_id is part of the primary key, and you will be all set.
